builtin/credential/cert/path_login.go

```diff
@@ -77,16 +78,21 @@ func (b *backend) pathLogin(
 		ttl = b.System().DefaultLeaseTTL()
 	}
 
+	maxTTL := matched.Entry.MaxTTL
+	if maxTTL == 0 {
+		maxTTL = b.System().MaxLeaseTTL()
+	}
+
 	clientCerts := req.Connection.ConnState.PeerCertificates
 	if len(clientCerts) == 0 {
-		return logical.ErrorResponse("no client certificate found"), nil
+		return nil, fmt.Errorf("no client certificate found")
 	}
 	skid := base64.StdEncoding.EncodeToString(clientCerts[0].SubjectKeyId)
 	akid := base64.StdEncoding.EncodeToString(clientCerts[0].AuthorityKeyId)
 
-	// Generate a response
 	resp := &logical.Response{
 		Auth: &logical.Auth{
+			Period: matched.Entry.Period,
```

Comment:
This actually *isn't* handled automatically (possibly/probably it should) -- if a period is non-zero, you should explicitly set the TTL to the period after the other TTL machinations are in place.

Reply:
This is handled explicitly below now.

Reply:
No longer handled explicitly at the backend level, check is now performed in core after #3677